Geometry/HGCalMapping/src/HGCalDenseMappingTool.cc

```diff
@@ -0,0 +1,52 @@
+#include "Geometry/HGCalMapping/interface/HGCalDenseMappingTool.h"
+
+HGCalDenseMappingTool::HGCalDenseMappingTool(HGCalDenseMappingToolConfig config) : config_(config) {}
+
+uint32_t HGCalDenseMappingTool::denseIndex(uint32_t sLink, uint32_t captureBlock) {
+  uint32_t rtn = sLink;
+  rtn = rtn * config_.sLinkCaptureBlockMax + captureBlock;
+  return rtn;
+}
+
+uint32_t HGCalDenseMappingTool::denseIndex(uint32_t sLink, uint32_t captureBlock, uint32_t eCOND) {
+  uint32_t rtn = sLink;
+  rtn = rtn * config_.sLinkCaptureBlockMax + captureBlock;
+  rtn = rtn * config_.captureBlockECONDMax + eCOND;
+  return rtn;
+}
+
+uint32_t HGCalDenseMappingTool::denseIndex(uint32_t sLink, uint32_t captureBlock, uint32_t eCOND, uint32_t eRx) {
+  uint32_t rtn = sLink;
+  rtn = rtn * config_.sLinkCaptureBlockMax + captureBlock;
+  rtn = rtn * config_.captureBlockECONDMax + eCOND;
+  rtn = rtn * config_.econdERXMax + eRx;
+  return rtn;
+}
+
+uint32_t HGCalDenseMappingTool::denseIndex(
+    uint32_t sLink, uint32_t captureBlock, uint32_t eCOND, uint32_t eRx, uint32_t channel) {
+  uint32_t rtn = sLink;
+  rtn = rtn * config_.sLinkCaptureBlockMax + captureBlock;
+  rtn = rtn * config_.captureBlockECONDMax + eCOND;
+  rtn = rtn * config_.econdERXMax + eRx;
+  rtn = rtn * config_.erxChannelMax + channel;
+  return rtn;
+}
+
+uint32_t HGCalDenseMappingTool::denseIndex(HGCalElectronicsId elecID) {
+  return denseIndex(
+      elecID.fedId(), elecID.captureBlock(), elecID.econdIdx(), elecID.econdeRx(), elecID.halfrocChannel());
+}
+
+HGCalElectronicsId HGCalDenseMappingTool::inverseDenseMap(uint32_t denseIdx) {
```

Comment:
Final suggestion
- name this function as inverseDenseMap
- fedId's will be number 0......N-1 for the negative endcap and N....2N-1 for the positive endcap so one could have another field in HGCalDenseMappingToolConfig called maxFEDsPerEndcap (we don't yet know the number but i would suggest setting it to 512 for the moment)
With this the sized would be s.th. like  `zside = fedId>config_.maxFEDsPerEndcap ? 1 : -1`